acceptance-tests/pom.xml

```diff
@@ -94,27 +101,54 @@
     </pluginRepositories>
 
     <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-failsafe-plugin</artifactId>
+                    <version>3.0.0-M4</version>
+                    <configuration>
+                        <trimStackTrace>false</trimStackTrace>
+                        <reportsDirectory>${project.build.directory}/surefire-reports</reportsDirectory>
+                        <reuseForks>false</reuseForks>
+                        <environmentVariables>
+                            <!-- Used by Jenkins Acceptance Test Harness -->
+                            <JENKINS_VERSION>${jenkins.version}</JENKINS_VERSION>
+                            <!-- The path to the plugin under test (i.e. the Bitbucket Server integration plugin) -->
+                            <LOCAL_JARS>${plugin.build.dir}/${plugin-under-test.name}.hpi</LOCAL_JARS>
+                            <!-- Always update the (local) plugins with the latest snapshot. This is especially useful if
+                            running the tests against an existing (external) Jenkins instance (see readme.md). -->
+                            <LOCAL_SNAPSHOTS>true</LOCAL_SNAPSHOTS>
+                            <!-- This runs Firefox inside a container. To be able to see the browser when running the test
+                            locally, change it to 'firefox' or 'chrome' (or whatever browser of choice).
+                            See https://github.com/jenkinsci/acceptance-test-harness/blob/master/docs/BROWSER.md for more
+                            details and the full list of browsers. -->
+                            <BROWSER>firefox-container</BROWSER>
```

Comment:
If I want to run this locally without Docker, can I override this without needing a file change?  If not, maybe we should put `firefox-container` into a property so it can be easily overridden on the command line (`-Dbrowser=firefox` for example?)

Also, the BbS tests run without Docker by default to optimize for developer experience. I've got a loose preference for that approach (so the Bamboo build would enable Docker but the default wouldn't use it).

Also, I couldn't get these tests to run on my machine at all with `firefox-container` (running Docker first, of course), `firefox` or `chrome`. I disabled my firewall but the Docker version couldn't connect. And the two "real browser" versions seem to need something downloaded. Do Firefox/Chrome (without Docker) work on your machine?

Error running the tests with BROWSER `filefox-container`:
```
[ERROR] it.com.atlassian.bitbucket.jenkins.internal.SmokeTest.testMultiBranchJobBitbucketTriggersReIndex  Time elapsed: 7.064 s  <<< ERROR!
com.google.inject.ProvisionException: 
Unable to provision, see the following errors:

1) Error in custom provider, org.openqa.selenium.remote.UnreachableBrowserException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
Build info: version: '3.14.0', revision: 'aacccce0', time: '2018-08-02T20:19:58.91Z'
System info: host: 'bplump-atlassian.local', ip: 'fe80:0:0:0:1c8d:e3d6:57e9:2b59%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.15.5', java.version: '1.8.0_221'
Driver info: driver.version: RemoteWebDriver
  at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:292) (via modules: org.jenkinsci.test.acceptance.guice.World -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.jenkinsci.test.acceptance.FallbackConfig)
  at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:292) (via modules: org.jenkinsci.test.acceptance.guice.World -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.jenkinsci.test.acceptance.FallbackConfig)
  while locating org.openqa.selenium.WebDriver
    for field at org.jenkinsci.test.acceptance.junit.DiagnosticRule.driver(DiagnosticRule.java:18)
  while locating org.jenkinsci.test.acceptance.junit.DiagnosticRule

1 error
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1025)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1051)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.addRule(JenkinsAcceptanceTestRule.java:151)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.collectGlobalRules(JenkinsAcceptanceTestRule.java:125)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.decorateWithRules(JenkinsAcceptanceTestRule.java:92)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.evaluate(JenkinsAcceptanceTestRule.java:58)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
Caused by: org.openqa.selenium.remote.UnreachableBrowserException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
Build info: version: '3.14.0', revision: 'aacccce0', time: '2018-08-02T20:19:58.91Z'
System info: host: 'bplump-atlassian.local', ip: 'fe80:0:0:0:1c8d:e3d6:57e9:2b59%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.15.5', java.version: '1.8.0_221'
Driver info: driver.version: RemoteWebDriver
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:569)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:212)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:130)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:143)
	at org.jenkinsci.test.acceptance.FallbackConfig.createContainerWebDriver(FallbackConfig.java:230)
	at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:132)
	at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:292)
	at org.jenkinsci.test.acceptance.FallbackConfig$$FastClassByGuice$$1b1c975b.invoke(<generated>)
	at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:272)
	at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:172)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision(InternalFactoryToInitializableAdapter.java:53)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(InternalFactoryToInitializableAdapter.java:45)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at org.jenkinsci.test.acceptance.guice.TestLifecycle$1.get(TestLifecycle.java:61)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:54)
	at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:132)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:85)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:267)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
	... 24 more
Caused by: java.net.ConnectException: Failed to connect to /127.0.0.1:56922
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:242)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:160)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:257)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:126)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
	at okhttp3.RealCall.execute(RealCall.java:77)
	at org.openqa.selenium.remote.internal.OkHttpClient.execute(OkHttpClient.java:105)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:101)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:73)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:136)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:548)
	... 50 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.java:129)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:240)
	... 72 more
```

Error running the tests with BROWSER `firefox`:
```
[ERROR] it.com.atlassian.bitbucket.jenkins.internal.SmokeTest.testMultiBranchJobManualReIndexing  Time elapsed: 1.167 s  <<< ERROR!
com.google.inject.ProvisionException: 
Unable to provision, see the following errors:

1) Error in custom provider, java.lang.IllegalStateException: The path to the driver executable must be set by the webdriver.gecko.driver system property; for more information, see https://github.com/mozilla/geckodriver. The latest version can be downloaded from https://github.com/mozilla/geckodriver/releases
  at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:292) (via modules: org.jenkinsci.test.acceptance.guice.World -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.jenkinsci.test.acceptance.FallbackConfig)
  at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:292) (via modules: org.jenkinsci.test.acceptance.guice.World -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.jenkinsci.test.acceptance.FallbackConfig)
  while locating org.openqa.selenium.WebDriver
    for field at org.jenkinsci.test.acceptance.junit.DiagnosticRule.driver(DiagnosticRule.java:18)
  while locating org.jenkinsci.test.acceptance.junit.DiagnosticRule

1 error
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1025)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1051)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.addRule(JenkinsAcceptanceTestRule.java:151)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.collectGlobalRules(JenkinsAcceptanceTestRule.java:125)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.decorateWithRules(JenkinsAcceptanceTestRule.java:92)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.evaluate(JenkinsAcceptanceTestRule.java:58)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
Caused by: java.lang.IllegalStateException: The path to the driver executable must be set by the webdriver.gecko.driver system property; for more information, see https://github.com/mozilla/geckodriver. The latest version can be downloaded from https://github.com/mozilla/geckodriver/releases
	at com.google.common.base.Preconditions.checkState(Preconditions.java:847)
	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:125)
	at org.openqa.selenium.firefox.GeckoDriverService.access$100(GeckoDriverService.java:43)
	at org.openqa.selenium.firefox.GeckoDriverService$Builder.findDefaultExecutable(GeckoDriverService.java:168)
	at org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:346)
	at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:129)
	at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:292)
	at org.jenkinsci.test.acceptance.FallbackConfig$$FastClassByGuice$$1b1c975b.invoke(<generated>)
	at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:272)
	at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:172)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision(InternalFactoryToInitializableAdapter.java:53)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(InternalFactoryToInitializableAdapter.java:45)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at org.jenkinsci.test.acceptance.guice.TestLifecycle$1.get(TestLifecycle.java:61)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:54)
	at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:132)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:85)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:267)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
	... 24 more
```

Error running the tests with BROWSER `chrome`:
```
[ERROR] it.com.atlassian.bitbucket.jenkins.internal.SmokeTest.testMultiBranchJobBitbucketTriggersReIndex  Time elapsed: 0.016 s  <<< ERROR!
com.google.inject.ProvisionException: 
Unable to provision, see the following errors:

1) Error in custom provider, java.lang.IllegalStateException: The path to the driver executable must be set by the webdriver.chrome.driver system property; for more information, see https://github.com/SeleniumHQ/selenium/wiki/ChromeDriver. The latest version can be downloaded from http://chromedriver.storage.googleapis.com/index.html
  at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:292) (via modules: org.jenkinsci.test.acceptance.guice.World -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.jenkinsci.test.acceptance.FallbackConfig)
  at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:292) (via modules: org.jenkinsci.test.acceptance.guice.World -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> org.jenkinsci.test.acceptance.FallbackConfig)
  while locating org.openqa.selenium.WebDriver
    for field at org.jenkinsci.test.acceptance.junit.DiagnosticRule.driver(DiagnosticRule.java:18)
  while locating org.jenkinsci.test.acceptance.junit.DiagnosticRule

1 error
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1025)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1051)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.addRule(JenkinsAcceptanceTestRule.java:151)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.collectGlobalRules(JenkinsAcceptanceTestRule.java:125)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.decorateWithRules(JenkinsAcceptanceTestRule.java:92)
	at org.jenkinsci.test.acceptance.junit.JenkinsAcceptanceTestRule$1.evaluate(JenkinsAcceptanceTestRule.java:58)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
Caused by: java.lang.IllegalStateException: The path to the driver executable must be set by the webdriver.chrome.driver system property; for more information, see https://github.com/SeleniumHQ/selenium/wiki/ChromeDriver. The latest version can be downloaded from http://chromedriver.storage.googleapis.com/index.html
	at com.google.common.base.Preconditions.checkState(Preconditions.java:847)
	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:125)
	at org.openqa.selenium.chrome.ChromeDriverService.access$000(ChromeDriverService.java:35)
	at org.openqa.selenium.chrome.ChromeDriverService$Builder.findDefaultExecutable(ChromeDriverService.java:156)
	at org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:346)
	at org.openqa.selenium.chrome.ChromeDriverService.createDefaultService(ChromeDriverService.java:91)
	at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:157)
	at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:149)
	at org.jenkinsci.test.acceptance.FallbackConfig.createWebDriver(FallbackConfig.java:292)
	at org.jenkinsci.test.acceptance.FallbackConfig$$FastClassByGuice$$1b1c975b.invoke(<generated>)
	at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:272)
	at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:172)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision(InternalFactoryToInitializableAdapter.java:53)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(InternalFactoryToInitializableAdapter.java:45)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at org.jenkinsci.test.acceptance.guice.TestLifecycle$1.get(TestLifecycle.java:61)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:54)
	at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:132)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:85)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:267)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
	... 24 more
```

Reply:
Interesting. I can run the tests in multiple ways (command-line Maven + from Intellij), although yes, I had to download the Gecko driver and point to it when running from Intellij (with _real_ Firefox for example). For Chrome, probably because I have Chromium installed, it doesn't ask for a driver to connect to the (remote) browser. I have also run it against a Firefox Selenium docker container by providing the remote browser URL, etc.
I should add more details to the README, especially for running from IDE perhaps.
These tests, to the best of knowledge, have only been run on Linux (CI + my machine + Martin's), so I _really_ hope they'll work on Macs too. :crossed_fingers: 
Let me add some more details here for now and please try again. Brb

Reply:
Obviously, if running from IDEA, need to have a Bitbucket instance running, then this is the run config I'm using:
![image](https://user-images.githubusercontent.com/32830630/87384834-08404f80-c5e0-11ea-9838-e16f73651073.png)

just running the tests from cmdline (from inside `acceptance-tests` dir) works for me too:
```
cd acceptance-tests
mvn clean verify -Psmoke-tests
```
Again, :notsureif: a Linux vs. Mac thing. Give the above IDEA config a try first?

Reply:
P.S. I _do_ have docker service running when I run the tests from the cmdline. The Jenkins ATH framework will spin up a Firefox Selenium container thingy for you, but it'll need the docker service to be running.

Reply:
- [ ] add more detailed instruction on how to use the non-container Firefox/Chrome with a WebDriver (e.g. Gecko) driver, etc.

Reply:
Yeah, when running with the container firefox, I had Docker running (I got a different error indicating it couldn't communicate with Docker if I didn't). 

I'll be interested to see if any other Mac folks can run this... that said, if Kristy the only other Mac user here, maybe it's not worth it.  (I'm not sure what Gaurav uses)

Reply:
I wanted to see how the browser install/startup works in BbS and it turns out we have a bit of magic called [atlassian-browsers](https://bitbucket.org/atlassian/atlassian-browsers/src/master/) taking care of downloading the binary (used [here](https://bitbucket.org/atlassian/atlassian-selenium/src/1cf62ee7f4b1fb1f44497bc5375b02abb9164f15/atlassian-webdriver/atlassian-webdriver-core/src/main/java/com/atlassian/webdriver/browsers/AutoInstallConfiguration.java#lines-29)).

No worries for now. Just a pointer to the `webdriver.gecko.driver` property in the README is good enough.

Reply:
I can't run the test on my mac machine as well. Same error as @bplump.

Reply:
- [ ] add more detailed instructions in README on how to use the non-container Firefox/Chrome with a WebDriver (e.g. Gecko) driver, etc.

- [ ] command-line test execution for MacOS is broken

Reply:
I've pushed some new instructions to the readme. Resolving this.